src/clusterfuzz/_internal/cron/grouper.py

```diff
@@ -97,19 +97,26 @@ def _emit_grouping_event(moved_testcase: int,
           group_merge_reason=group_merge_reason))
 
 
+def _get_testcase_log_info(testcase: TestcaseAttributes):
+  """Return testcase data as string for logging."""
+  log_str = (f'group={testcase.group_id}, '
```

Comment:
I can be missing the entire context, but could it be an structured log?

Reply:
Besides the performance purposes for structure logs, I think it would make the code cleaner. Maybe to simplify you can just set a namedtuple with these fields and pass it as extras to the log call. wdyt?